src/supautils.c

```diff
@@ -310,7 +307,10 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 					break;
 				}
 
-				comfirm_reserved_roles(get_rolespec_name(stmt->role));
+				if (!has_privileged_role()){
+					confirm_reserved_roles(get_rolespec_name(stmt->role));
+					break;
+				}
```

Comment:
This means `privileged_role` can set allowed_configs for reserved roles, incl. the superuser, which may have unintended effects (e.g. setting `pgaudit.*` for internal maintenance queries).

Can we have a separate config `configurable_reserved_roles` instead? Ones that you can `alter role set` but can't `drop` etc.

FYI [here's](https://github.com/supabase/postgres/blob/develop/ansible/files/postgresql_config/supautils.conf.j2) the current supautils config we use.

Reply:
> This means privileged_role can set allowed_configs for reserved roles, incl. the superuser

@soedirgo How about instead banning the ability to configure superusers placeholders GUCs?

This already fails under `privileged_role`:

```bash
$ supautils-with-pg-15 psql -U privileged_role
```

```sql
alter role postgres set search_path to public;
ERROR:  must be superuser to alter superusers
```

Placeholders are a special case that need to be handled:

```sql
alter role postgres set pgrst.qwer to true;
ALTER ROLE
```

Reply:
> Can we have a separate config configurable_reserved_roles instead? Ones that you can alter role set but can't drop etc.

Hm, more configs increases complexity. Another option could be to mark configurable reserved roles with `<role*`>`, like:

```
supautils.reserved_roles = 'supabase_admin, anon*, supabase_auth_admin'
```

WDYT?

Reply:
Pardon the late response.

> How about instead banning the ability to configure superusers placeholders GUCs? Would that be enough?

I think it's better to be on the conservative side of being restrictive, otherwise if we decide to restrict for other reserved roles, there'll be a lot of cleanup work (issuing notifications, etc.).

> Another option could be to mark configurable reserved roles with <role*>`, like

I don't really mind either approach (one is more config, the other needs more parsing work).

Reply:
> Another option could be to mark configurable reserved roles with <role*>

@soedirgo I went with the above option, LMWYT.